Documentation improvements

zackgalbreath opened the pull request:
* Clarify documentation for dependency management with CMake
* Improve FetchContent example in docs

@jtattermusch

jtattermusch commented on BUILDING.md:
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
+that are already installed on your system and use them to build gRPC.
+
+This behavior is controlled by the `gRPC_<depname>_PROVIDER` CMake variables,
+ie `gRPC_CARES_PROVIDER`. The options that these variables take are as follows:
```
nit: s/ie/e.g./ ?

zackgalbreath replied:
👍

jtattermusch commented on BUILDING.md:
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
+ie `gRPC_CARES_PROVIDER`. The options that these variables take are as follows:
+
 * module - build dependencies alongside gRPC.
 * package - use external copies of dependencies that are already available
```
nit: add "(preferrably you'd pre-install these also using cmake, with the `CMAKE_INSTALL_PREFIX` option)" ?

zackgalbreath replied:
👍

jtattermusch commented on BUILDING.md:
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
+This behavior is controlled by the `gRPC_<depname>_PROVIDER` CMake variables,
+ie `gRPC_CARES_PROVIDER`. The options that these variables take are as follows:
+
 * module - build dependencies alongside gRPC.
```
nit: add "(the source code will be obtained from the git submodules)"?

zackgalbreath replied:
👍

jtattermusch commented on src/cpp/README.md:
```diff
@@ -70,18 +70,26 @@ also sets up an `add_subdirectory()` rule for you. This causes gRPC to be
 FetchContent_MakeAvailable(gRPC)
 
-add_executable(my_exe my_exe.cc)
+add_executable(my_exe my_exe.cxx)
```
why cxx and not .cc? (this repo uses .cc for C++ files)

zackgalbreath replied:
👍

zackgalbreath commented:
Thanks for the feedback, @jtattermusch . I believe I've addressed your concerns. Please let me know if you'd like to see any more changes before we merge this.

zackgalbreath commented:
@jtattermusch can we merge this?

jtattermusch commented:
CC @chalin  FYI

jtattermusch commented:
@

> @jtattermusch can we merge this?

Sure. Sorry, this feel through the cracks.
